src/storage/src/healthcheck.rs

```diff
@@ -543,10 +564,15 @@ impl HealthStatusUpdate {
         }
     }
 
+    /// Generates a ceasing [`HealthStatusUpdate`] with `update`.
+    pub(crate) fn ceasing(error: String) -> Self {
+        HealthStatusUpdate::Ceased { error }
+    }
+
     /// Whether or not we should halt the dataflow instances and restart it.
     pub(crate) fn should_halt(&self) -> bool {
         match self {
-            HealthStatusUpdate::Running => false,
+            HealthStatusUpdate::Running | HealthStatusUpdate::Ceased { .. } => false,
```

Comment:
so, do we assume that the source impl itself shuts itself down?

Reply:
might be worth a comment!

Reply:
yep!